specs/dynamicdependencies/DynamicDependencies.md

```diff
@@ -0,0 +1,1256 @@
+// minVersion, and packageDependencyProcessorArchitectures values.
+STDAPI_(BOOL) MddArePackageDependencyIdsEquivalent(
+    _In_ PCWSTR packageDependencyId1,
+    _In_ PCWSTR packageDependencyId2);
+
+// Return TRUE if packageDependencyIdContext1 and packageDependencyContextId2
+// are associated with the same resolved package.
+STDAPI_(BOOL) MddArePackageDependencyContextsEquivalent(
+    _In_ const MDD_PACKAGEDEPENDENCY_CONTEXT* packageDependencyContext1,
+    _In_ const MDD_PACKAGEDEPENDENCY_CONTEXT* packageDependencyContext2);
+```
+
+## 6.2. WinRT API
+
+```c# (but really MIDL3)
+namespace Microsoft.ApplicationModel
+{
+/// CPU architectures to optionally filter available packages against a package dependency.
+/// These generally correspond to processor architecture types supported by MSIX.
+/// @see Windows.System.ProcessorArchitecture
+enum PackageDependencyProcessorArchitectures
+{
+    None    = 0,
+    Neutral = 0x00000001,
+    X86     = 0x00000002,
+    X64     = 0x00000004,
+    Arm     = 0x00000008,
+    Arm64   = 0x00000010,
+    X86A64  = 0x00000020,
+};
+
+/// Options when 'pinning' a package dependency
+runtimeclass PinPackageDependencyOptions
+{
+    PinPackageDependencyOptions();
+
+    /// Optional filtering by cpu architecture(s)
+    PackageDependencyProcessorArchitectures Architectures;
+
+    /// Do not verify at least 1 matching package exists when pinning a package dependency
+    boolean DoNotVerifyDependencyResolution;
+
+    /// An absolute filename or path to use as a lifetime artifact when pinning a package dependency.
+    /// If set, the package dependency is implicitly unpinned when this is deleted.
+    String LifetimeArtifactFileOrPath;
+
+    /// A registry key to use as a lifetime artifact when pinning pinning a package dependency.
+    /// If set, the package dependency is implicitly unpinned when this is deleted.
+    String LifetimeArtifactRegistrySubkey;
+}
+
+/// Options when adding a package dependency
+runtimeclass AddPackageDependencyOptions
+{
+    AddPackageDependencyOptions();
+
+    /// The rank when adding the package dependency to a a package graph.
+    /// @note A package graph is sorted in ascending order from -infinity...0...+infinity
+    /// @note The default value is zero
+    int Rank;
+
+    /// If a package dependency resolves to a package with dependencies and this option is true
+    /// the dependencies are ignored; only the package is added to the package graph. By default
+    /// a resolved package and its dependencies are added to the package graph.
+    boolean OnlyUseFirstPackageFamily;
+
+    /// If a package is added to a package graph with a package of the same rank (aka a collision on rank)
+    /// and this option is true the package is prepended to the set of packages of the same rank.
+    /// By default packages are appended to the set of packages with the same rank.
+    boolean PrependIfRankCollision;
+}
+
+/// TBD
+runtimeclass PackageDependency
```

Comment:
Can I use this WinRT API from my app before I add a reference to the Project Reunion framework package? Or is this only for interacting with the PR-based dependency manager types?

Reply:
You can use the Dynamic Dependencies API after the 'bootstrapper API' has been called.

See 5.4. Packaging - ProjectReunion.dll and ProjectReunion.Loader.dll for more details